persistence/src/main/java/edu/unc/lib/dl/persist/services/edit/ExpireEmbargoService.java

```diff
@@ -0,0 +1,187 @@
+
+        if (resourceList.size() > 0) {
+            // remove all expired embargoes
+            for (String rescUri: resourceList) {
+                FedoraTransaction tx = txManager.startTransaction();
+
+                try (Timer.Context context = timer.time()) {
+                    PID pid = PIDs.get(rescUri);
+                    currentPid = pid;
+                    RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
+                    Resource resc = repoObj.getResource();
+
+                    // remove embargo
+                    String embargoDate = resc.getProperty(embargoUntil).getString();
+                    repoObjFactory.deleteProperty(repoObj, embargoUntil);
+                    pids.add(pid);
+                    String eventText = "Expired an embargo which ended " +
+                            formatDateToUTC(parseUTCToDate(embargoDate));
+                    // Produce the premis event for this embargo
+                    repoObj.getPremisLog().buildEvent(Premis.Dissemination)
+                            .addSoftwareAgent(SoftwareAgent.embargoExpirationService.getFullname())
+                            .addEventDetail(eventText)
+                            .writeAndClose();
+                } catch (Exception e) {
+                    tx.cancelAndIgnore();
+                    log.error("Failed to expire embargo for {} with error:", currentPid, e);
+                } finally {
+                    tx.close();
+                }
+            }
+        } else {
+            log.error("No embargoes to expire");
+        }
+
+        if (!pids.isEmpty()) {
+            // send a message for expired embargoes
+            operationsMessageSender.sendOperationMessage(SoftwareAgent.embargoExpirationService.getFullname(),
+                    JMSMessageUtil.CDRActions.EDIT_ACCESS_CONTROL,
+                    pids);
+        }
+    }
+
+    private final static String EMBARGO_QUERY =
+            "PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n" +
+                    "select ?resource\n" +
+                    "where {\n" +
+                    "  ?resource <http://cdr.unc.edu/definitions/acl#embargoUntil> ?date .\n" +
+                    "  FILTER (?date < \"%s\"^^xsd:dateTime)\n" +
+                    "}";
+
+    private List<String> getEmbargoInfo() {
+        String today = DateTimeUtil.formatDateToUTC(new Date());
+        String query = String.format(EMBARGO_QUERY, today);
+
+        try (QueryExecution exec = sparqlQueryService.executeQuery(query)) {
+            ResultSet resultSet = exec.execSelect();
+            List<String> embargoedRescList = new ArrayList<>();
+
+            for (; resultSet.hasNext() ;) {
+                QuerySolution soln = resultSet.nextSolution();
+                Resource resc = soln.getResource("resource");
+                embargoedRescList.add(resc.getURI());
+            }
+            return embargoedRescList;
+        } catch (NullPointerException e) {
```

Comment:
Its not all that common to want to catch a NullPointerException. Which part of this is throwing one? Would it be possible to check whatever object it is that is triggering it to make sure its not null first?